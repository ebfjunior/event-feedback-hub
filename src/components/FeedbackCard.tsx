import { Star } from 'lucide-react';
import { cn } from '@/lib/utils';
import type { FeedbackItem } from '@/types/api';

export type FeedbackCardProps = {
  feedback: FeedbackItem;
  highlight?: boolean;
};

function Stars({ rating }: { rating: number }) {
  return (
    <div className="flex items-center gap-1" aria-label={`${rating} out of 5 stars`}>
      {Array.from({ length: 5 }).map((_, i) => (
        <Star
          key={i}
          className={cn('h-4 w-4', i < rating ? 'fill-yellow-400 stroke-yellow-400' : 'stroke-muted-foreground')}
        />
      ))}
    </div>
  );
}

export function FeedbackCard({ feedback, highlight }: FeedbackCardProps) {
  const created = new Date(feedback.created_at);
  const timeAgo = new Intl.RelativeTimeFormat(undefined, { numeric: 'auto' });
  const diffMs = Date.now() - created.getTime();
  const diffMin = Math.round(diffMs / 60000);
  const rtf = diffMin < 60 ? timeAgo.format(-diffMin, 'minute') : timeAgo.format(-Math.round(diffMin / 60), 'hour');

  return (
    <div className={cn('rounded-md border bg-card/60 p-3 shadow-sm', highlight && 'ring-1 ring-yellow-400')}> 
      <div className="mb-1 flex items-center justify-between text-xs text-muted-foreground">
        <div className="truncate font-medium text-primary/80">{feedback.event_name}</div>
        <div>{rtf}</div>
      </div>
      <div className="mb-2 text-sm text-foreground/90">{feedback.text}</div>
      <div className="flex items-center justify-between">
        <div className="text-xs text-muted-foreground">Anonymous</div>
        <Stars rating={feedback.rating} />
      </div>
    </div>
  );
}
