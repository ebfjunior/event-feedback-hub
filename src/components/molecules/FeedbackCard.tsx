import Link from 'next/link';
import type { FeedbackItem } from '@/types/api';
import { cn } from '@/lib/utils';
import { Card, CardContent, CardDescription, CardHeader, CardAction } from '@/components/ui/card';
import { Badge } from '@/components/ui/badge';
import { Stars } from '@/components/atoms/Stars';

export type FeedbackCardProps = {
  feedback: FeedbackItem;
  highlight?: boolean;
};

export function FeedbackCard({ feedback, highlight }: FeedbackCardProps) {
  const created = new Date(feedback.created_at);
  const timeAgo = new Intl.RelativeTimeFormat(undefined, { numeric: 'auto' });
  const diffMs = Date.now() - created.getTime();
  const diffMin = Math.round(diffMs / 60000);
  const rtf = diffMin < 60 ? timeAgo.format(-diffMin, 'minute') : timeAgo.format(-Math.round(diffMin / 60), 'hour');

  return (
    <Card className={cn('bg-card/60 shadow-sm', highlight && 'ring-1 ring-yellow-400')}>
      <CardHeader className="gap-1.5">
        <Badge asChild variant="secondary" className="truncate max-w-full">
          <Link href={`/events/${feedback.event_id}`} aria-label={`Open event ${feedback.event_name}`}>
            {feedback.event_name}
          </Link>
        </Badge>
        <CardDescription className="text-xs">{rtf}</CardDescription>
        <CardAction>
          <Stars rating={feedback.rating} />
        </CardAction>
      </CardHeader>
      <CardContent className="pt-0">
        <div className="text-sm text-foreground/90">{feedback.text}</div>
        <div className="mt-3 text-xs text-muted-foreground">Anonymous</div>
      </CardContent>
    </Card>
  );
}


