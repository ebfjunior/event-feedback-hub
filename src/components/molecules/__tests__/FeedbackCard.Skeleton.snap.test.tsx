import React from 'react';
import { render } from '@testing-library/react';
import { FeedbackCardSkeleton } from '../FeedbackCard.Skeleton';

describe('FeedbackCardSkeleton (snapshot)', () => {
	it('renders', () => {
		const { container } = render(<FeedbackCardSkeleton />);
		expect(container.firstChild).toMatchSnapshot();
	});
});


