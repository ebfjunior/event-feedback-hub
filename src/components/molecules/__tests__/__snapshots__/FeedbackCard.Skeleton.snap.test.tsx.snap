// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FeedbackCardSkeleton (snapshot) > renders 1`] = `
<div
  class="text-card-foreground flex flex-col gap-6 rounded-xl border py-6 shadow-sm animate-pulse bg-card/60"
  data-slot="card"
>
  <div
    class="@container/card-header grid auto-rows-min grid-rows-[auto_auto] items-start px-6 has-data-[slot=card-action]:grid-cols-[1fr_auto] [.border-b]:pb-6 gap-1.5"
    data-slot="card-header"
  >
    <div
      class="h-5 w-40 rounded bg-muted"
    />
    <div
      class="h-3 w-16 rounded bg-muted"
    />
    <div
      class="col-start-2 row-span-2 row-start-1 self-start justify-self-end"
      data-slot="card-action"
    >
      <div
        class="h-4 w-24 rounded bg-muted"
      />
    </div>
  </div>
  <div
    class="px-6 pt-0"
    data-slot="card-content"
  >
    <div
      class="space-y-2"
    >
      <div
        class="h-3 w-full rounded bg-muted"
      />
      <div
        class="h-3 w-5/6 rounded bg-muted"
      />
      <div
        class="h-3 w-4/6 rounded bg-muted"
      />
    </div>
    <div
      class="mt-3 h-3 w-24 rounded bg-muted"
    />
  </div>
</div>
`;
